packages/ui-components/src/components/InputGroup/InputGroup.component.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Juno contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { ReactNode, isValidElement } from "react"
+import { Stack } from "../Stack"
+import "./input-group.scss"
+
+type VariantTypes = "default" | "primary" | "primary-danger" | "subdued"
+
+export interface InputGroupProps {
+  /**
+   * The children to render within the InputGroup.
+   * This can be any React node or a collection of React nodes such as Buttons, TextInput, and Select elements.
+   */
+  children?: React.ReactNode
+  /**
+   * Additional CSS class name(s) to apply to the InputGroup for custom styling.
+   */
+  className?: string
+  /**
+   * The variant style to apply to the group and its children.
+   */
+  variant?: VariantTypes
+  /**
+   * If true, all elements within the InputGroup will be disabled.
+   * Individual elements can override this setting if needed.
+   */
+  disabled?: boolean
+}
+
+type OverrideProps = { variant?: VariantTypes; disabled?: boolean }
+
+/**
+ * Clones a child element with inherited variant and disabled props.
+ * Individual child props will override inherited ones if specified.
+ *
+ * @returns A cloned child element with the variant and disabled props applied.
+ */
+const cloneElementWithInheritedProps = (
+  child: ReactNode,
+  parentVariant: VariantTypes,
+  parentDisabled: boolean
+): ReactNode => {
+  if (!isValidElement<OverrideProps>(child)) return child
+
+  const combinedProps = {
+    variant: child.props.variant ?? parentVariant,
```

Comment:
What's your reason for replacing the `||` operator in the original component with the nullish coalescing operator? I think `||` is more appropriate here as it reacts to all falsy values not only `null` and `undefined`

In the future (not now) we need to revisit this in general. I think it makes more sense for the parent's prop value to always overwrite the child if it exists. But this component is still WiP for a reason, it's not completely finished